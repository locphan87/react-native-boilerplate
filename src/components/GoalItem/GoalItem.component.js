// @flow
import React from 'react'
import { ListItem } from 'react-native-elements'

import Progress from '../Progress/Progress.component'
import type { Goal } from '../../types/goal.type'

import styles from './GoalItem.component.style'

type Props = Goal & {
  onPress: Function
}

const leftIcon = {
  name: 'target',
  type: 'foundation',
  color: 'white'
}

const GoalItem = (props: Props) => {
  const { id, title, onPress, ...rest } = props

  return (
    <ListItem
      title={title}
      containerStyle={styles.container}
      titleStyle={styles.title}
      subtitle={<Progress {...rest} type={'simple'} />}
      subtitleStyle={styles.subtitle}
      underlayColor={'blue'}
      onPress={onPress(id)}
      leftIcon={leftIcon}
      chevron
      bottomDivider
    />
  )
}
GoalItem.displayName = 'GoalItem'

export default GoalItem
