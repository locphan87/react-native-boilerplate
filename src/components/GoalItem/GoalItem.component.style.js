// @flow
import { StyleSheet } from 'react-native'

import { COLORS } from '../../themes'

const styles = {
  container: {
    flex: 1,
    backgroundColor: 'transparent'
  },
  title: {
    color: COLORS.white
  },
  subtitle: {
    marginLeft: 5
  }
}

export default StyleSheet.create(styles)
