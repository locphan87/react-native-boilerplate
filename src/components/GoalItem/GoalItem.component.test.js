import { testSnapshots } from '../../utils/test.util'

import GoalItem from './GoalItem.component'

const props = {
  id: 1,
  title: 'title',
  start: 1,
  current: 1,
  end: 2,
  onPress: id => jest.fn()
}

describe('GoalItem', () => {
  testSnapshots(GoalItem, [
    {
      props,
      description: 'basic render'
    }
  ])
})
