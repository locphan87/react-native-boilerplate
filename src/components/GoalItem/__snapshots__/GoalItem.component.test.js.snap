// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalItem snapshots basic render 1`] = `
<RNE_ListItem
  bottomDivider={true}
  chevron={true}
  containerStyle={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  leftIcon={
    Object {
      "color": "white",
      "name": "target",
      "type": "foundation",
    }
  }
  onPress={[MockFunction]}
  subtitle={
    <Progress
      current={1}
      end={2}
      start={1}
      type="simple"
    />
  }
  subtitleStyle={
    Object {
      "marginLeft": 5,
    }
  }
  title="title"
  titleStyle={
    Object {
      "color": "white",
    }
  }
  underlayColor="blue"
/>
`;
